File: 1603-running-sum-of-1d-array/1603-running-sum-of-1d-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
 int left(int a[], int n, int x)
 {
    int c=0;
    for(int i=0; i<=x; i++){
        c+=a[i];
    }
    return c;
 }
int* runningSum(int* nums, int numsSize, int* returnSize) {
    int *res = (int*)malloc(numsSize*sizeof(int));
    for(int i=0; i<numsSize; i++)
    {
        res[i] = left(nums, numsSize, i);
    }
    *returnSize = numsSize;
    return res;
}

File: 1603-running-sum-of-1d-array/1603-running-sum-of-1d-array.java
class Solution {
    static int left(int[] a, int i){
        int sum=0;
        for(int j=0; j<=i; j++) sum+=a[j];
        return sum;
    }
    public int[] runningSum(int[] nums) {
        int[] res = new int[nums.length];
        for(int i=0; i<nums.length; i++){
            res[i] = left(nums, i);
        }
        return res;
    }
}